src/test/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMNavigatorTest.java

```diff
@@ -145,37 +146,45 @@ public void prepareMockGitHub() throws Exception {
 
     @Test
     public void fetchSmokes() throws Exception {
-        final LogTaskListener listener = new LogTaskListener(Logger.getAnonymousLogger(), Level.INFO);
```

Comment:
The `fetchSmokes()` test should be the testing the same things. I just pulled out the logic for creating `SCMSourceObserver` into another method called `getObserver` because I wanted to reuse the same bit of code.

I also added other unit tests here like `fetchReposFromTeamSlug` to check the trait is used to filter projects based on the team name and added the appropriate json response.

Hopefully that makes a little more sense. I'm happy to revert back the `fetchSmokes()` if the diff is too confusing.